File: src/main/java/com/duneyrefrigeracao/backend/domain/model/Servico.java
package com.duneyrefrigeracao.backend.domain.model;

import com.duneyrefrigeracao.backend.domain.enums.StatusServico;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Date;

@Entity
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Servico {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String descricao;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @Enumerated(EnumType.STRING)
    private StatusServico statusServico;

    @Column(nullable = false)
    private Date dtCriacao;

    @Column(nullable = false)
    private Date dtInicial;

    @Column
    private Date dtFinalizacao;
}

File: frontend/src/dto/servico/Servico.ts
import Cliente from "../cliente/Cliente"
import FornecedorServico from "./FornecedorServico"
import ProdutoServico from "./ProdutoServico"
import TecnicoServico from "./TecnicoServico"

type Servico = {
    id: number,
    descricao: string,
    statusServico: string,
    dtInicial: string,
    dtFinalizacao: string,
    dtCriacao: string,
    cliente: Cliente,
    listaProduto: ProdutoServico[],
    listaTecnico: TecnicoServico[],
    listaFornecedor: FornecedorServico[]

}

export default Servico;